### PR TITLE
fix Element.shadow example bug

### LINKED ISSUE #13454: Issue with "Element.attachShadow()": small bug in the example
MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow

#### What information was incorrect, unhelpful, or incomplete?
In the example, the `countWords(node)` function counts words wrong. For empty string it counts 1 and for string with only new line character it counts two words. 

#### Specific section or headline?
https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow#examples
and also
https://mdn.github.io/web-components-examples/word-count-web-component/

#### What did you expect to see?
1. It needs to return 0 for empty string. We can use `filter` function like this;
```
function countWords(node) {
    var text = node.innerText || node.textContent
    return text.trim().split(/\s+/g).filter(a => a.trim().length > 0).length;
}
```
2. Fix the linked live example: https://github.com/mdn/web-components-examples/tree/master/word-count-web-component
3. Also, the live example code has been [improved](https://github.com/mdn/web-components-examples/commit/690839c536e367312d88f84be129873d2cc53227). So update the code in this post accordingly.

#### Did you test this? If so, how?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/element/attachshadow`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/element/attachshadow/index.md
* Last commit: https://github.com/mdn/content/commit/89c8199b06e7fb2b76da4087cb9eedef065b2506
* Document last modified: 2022-02-18T09:24:31.000Z

</details>

### PR DESCRIPTION
**Pull request by b0nbon1**
#### Summary

[Fixes issue #13454]( https://github.com/mdn/content/issues/13454) - fixes function `countWords` where for empty strings it returns 0 instead of 1.

#### Motivation
function `countWords returns 2 when there is an empty an empty string.

#### Related issues
[fixes #13454]( https://github.com/mdn/content/issues/13454)
[Linked PR on web components](https://github.com/mdn/web-components-examples/pull/39)

#### Metadata

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by jpmedley** [comment]
I'm curious, what problem does this solve?

**Review by wbamberg — APPROVED** [review]
👍 thank you @b0nbon1 !

File: files/en-us/web/api/customelementregistry/index.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ class WordCount extends HTMLParagraphElement {
 
     function countWords(node){
       var text = node.innerText || node.textContent
```
Comment (shenotme) [inline files/en-us/web/api/customelementregistry/index.md:44]: Error end

Comment (shenotme) [inline files/en-us/web/api/customelementregistry/index.md:44]: End error

File: files/en-us/web/api/customelementregistry/index.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ class WordCount extends HTMLParagraphElement {
 
     function countWords(node){
```
Comment (shenotme) [inline files/en-us/web/api/customelementregistry/index.md:43]: End

File: files/en-us/web/api/customelementregistry/index.md (hunk 1 above)
Comment (shenotme) [inline files/en-us/web/api/customelementregistry/index.md:44]: Error end

File: files/en-us/web/api/customelementregistry/index.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ class WordCount extends HTMLParagraphElement {
...
       var text = node.innerText || node.textContent
-      return text.split(/\s+/g).length;
```
Comment (shenotme) [inline files/en-us/web/api/customelementregistry/index.md:45]: Flase

File: files/en-us/web/api/customelementregistry/index.md [hunk 4]
```diff
@@ -42,7 +42,7 @@ class WordCount extends HTMLParagraphElement {
...
+      return text.trim().split(/\s+/g).filter(a => a.trim().length > 0).length;
```
Comment (shenotme) [inline files/en-us/web/api/customelementregistry/index.md:45]: Flase end

File: files/en-us/web/api/customelementregistry/index.md (hunk 3 above)
Comment (shenotme) [inline files/en-us/web/api/customelementregistry/index.md:45]: End result

File: files/en-us/web/api/element/attachshadow/index.md [hunk 5]
```diff
@@ -106,7 +106,7 @@ class WordCount extends HTMLParagraphElement {
 
     function countWords(node){
       var text = node.innerText || node.textContent
-      return text.trim().split(/\s+/g).length;
```
Comment (shenotme) [inline files/en-us/web/api/element/attachshadow/index.md:109]: End returns

File: files/en-us/web/api/customelementregistry/index.md [hunk 6]
```diff
@@ -42,7 +42,7 @@ class WordCount extends HTMLParagraphElement {
...
     }
 
     var count = 'Words: ' + countWords(wcParent);
```
Comment (shenotme) [inline files/en-us/web/api/customelementregistry/index.md:48]: Flase end

File: files/en-us/web/api/element/attachshadow/index.md (hunk 5 above)
Comment (shenotme) [inline files/en-us/web/api/element/attachshadow/index.md:109]: End

Comment (shenotme) [inline files/en-us/web/api/element/attachshadow/index.md:109]: Node

File: files/en-us/web/api/customelementregistry/index.md (hunk 6 above)
Comment (shenotme) [inline files/en-us/web/api/customelementregistry/index.md:48]: Error end

**Pull request merged**